sktime/transformations/series/bootstrapping.py

```diff
@@ -0,0 +1,401 @@
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Bootstrapping methods for time series."""
+
+__author__ = ["ltsaprounis"]
+
+from copy import copy
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
+from statsmodels.tsa.api import STL as _STL
+
+from sktime.transformations.base import BaseTransformer
+from sktime.transformations.series.boxcox import BoxCoxTransformer
+
+
+class BootsrappingTransformer(BaseTransformer):
+    """Creates a population of similar time series.
+
+    This method utilises a form of bootstrapping to generate a population of
+    similar time series to the input time series [1]_, [2]_.
+
+    First the observed time series is transformed using a Box-Cox transformation to
+    stabilise the variance. Then it's decomposed to seasonal, trend and residual
+    time series, using the STL implementation from statsmodels [4]_. We then sample
+    blocks from the residuals time series using the Moving Block Bootstrapping (MBB)
+    method [3]_ to create synthetic residuals series that mimic the autocorrelation
+    patterns of the observed series. Finally these bootstrapped residuals are added
+    to the season and trend components and we use the inverse Box-Cox transform to
+    return a panel of similar time series.
+
+    The resulting panel can be used for Bagging forecasts, prediction intervals and
+    data augmentation.
```

Comment:
the description should detail the structure of the returned data frame or panel object, since multiple variables are generated. Since the panel object could be `numpy`, the order of columns should also figure in this description.